clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy

```diff
@@ -80,6 +91,287 @@ abstract class OpenstackClientProvider {
+    handleRequest {
+      getRegionClient(region).networking().subnet().get(subnetId) != null
+    }
+  }
+
+  /**
+   * Creates load balancer pool in provided region.
+   * @param region
+   * @param loadBalancerPool
+   * @return LbPool
+   */
+  LbPool createLoadBalancerPool(final String region, final LoadBalancerPool loadBalancerPool) {
+    handleRequest {
+      getRegionClient(region).networking().loadbalancers().lbPool().create(
+        Builders.lbPool()
+          .name(loadBalancerPool.derivedName)
+          .protocol(Protocol.forValue(loadBalancerPool.protocol?.name()))
+          .lbMethod(LbMethod.forValue(loadBalancerPool.method?.name()))
+          .subnetId(loadBalancerPool.subnetId)
+          .description(loadBalancerPool.description)
+          .adminStateUp(Boolean.TRUE)
+          .build())
+    }
+  }
+
+  /**
+   * Updates existing load balancer pool's name or load balancer method.
+   * @param region
+   * @param loadBalancerPool
+   * @return
+     */
+  LbPool updateLoadBalancerPool(final String region, final LoadBalancerPool loadBalancerPool) {
+    handleRequest {
+      getRegionClient(region).networking().loadbalancers().lbPool().update(loadBalancerPool.id,
+        Builders.lbPoolUpdate()
+          .name(loadBalancerPool.derivedName)
+          .lbMethod(LbMethod.forValue(loadBalancerPool.method?.name()))
+          .description(loadBalancerPool.description)
+          .adminStateUp(Boolean.TRUE)
+          .build())
+    }
+  }
+
+  /**
+   * Creates VIP for given region and pool.
+   * @param region
+   * @param virtualIP
+   * @return
+   */
+  Vip createVip(final String region, final VirtualIP virtualIP) {
+    handleRequest {
+      getRegionClient(region).networking().loadbalancers().vip().create(
+        Builders.vip()
+          .name(virtualIP.derivedName)
+          .subnetId(virtualIP.subnetId)
+          .poolId(virtualIP.poolId)
+          .protocol(Protocol.forValue(virtualIP.protocol?.name()))
+          .protocolPort(virtualIP.port)
+          .adminStateUp(Boolean.TRUE)
+          .build())
+    }
+  }
+
+  /**
+   * Updates VIP in specified region.
+   * @param region
+   * @param virtualIP
+   * @return
+   */
+  Vip updateVip(final String region, final VirtualIP virtualIP) {
+    handleRequest {
+      //Currently only supporting updates to name ... Could be expanded to SessionPersistence & connectionLimit
+      getRegionClient(region).networking().loadbalancers().vip().update(virtualIP.id,
+        Builders.vipUpdate().name(virtualIP.derivedName).adminStateUp(Boolean.TRUE).build())
+    }
+  }
+
+  /**
+   * Gets HealthMonitor for given region and id.
+   * @param region
+   * @param healthMonitorId
+   * @return
+   */
+  HealthMonitor getHealthMonitor(final String region, final String healthMonitorId) {
+    HealthMonitor result = handleRequest {
+      getRegionClient(region).networking().loadbalancers().healthMonitor().get(healthMonitorId)
+    }
+    if (result == null) {
```

Comment:
I'll stop commenting on these, but the "groovy" way is to rely on groovy-truthiness